File: frameNet/src/main/java/org/sqlunet/framenet/FnSentencePointer.java
/*
 * Copyright (c) 2019. Bernard Bou <1313ou@gmail.com>.
 */

package org.sqlunet.framenet;

import android.os.Parcel;
import android.os.Parcelable;

import org.sqlunet.Pointer;

import androidx.annotation.NonNull;

/**
 * Parcelable sentence
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public class FnSentencePointer extends Pointer
{
	/**
	 * Static field used to regenerate object, individually or as arrays
	 */
	static public final Parcelable.Creator<FnSentencePointer> CREATOR = new Parcelable.Creator<FnSentencePointer>()
	{
		@Override
		public FnSentencePointer createFromParcel(@NonNull final Parcel parcel)
		{
			return new FnSentencePointer(parcel);
		}

		@Override
		public FnSentencePointer[] newArray(final int size)
		{
			return new FnSentencePointer[size];
		}
	};

	/**
	 * Constructor from parcel, reads back fields IN THE ORDER they were written
	 */
	private FnSentencePointer(@NonNull final Parcel parcel)
	{
		super(parcel);
	}

	/**
	 * Constructor
	 *
	 * @param sentenceId sentence id
	 */
	public FnSentencePointer(final long sentenceId)
	{
		super(sentenceId);
	}
}

File: frameNet/src/main/java/org/sqlunet/framenet/FnSentencePointer.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.framenet

import android.os.Parcel
import android.os.Parcelable
import org.sqlunet.Pointer

/**
 * Parcelable sentence
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
class FnSentencePointer : Pointer {

    /**
     * Constructor from parcel, reads back fields IN THE ORDER they were written
     */
    private constructor(parcel: Parcel) : super(parcel)

    /**
     * Constructor
     *
     * @param sentenceId sentence id
     */
    constructor(sentenceId: Long) : super(sentenceId)

    override fun describeContents(): Int {
        return 0
    }

    companion object {

        @JvmField
        val CREATOR = object : Parcelable.Creator<FnSentencePointer> {

            override fun createFromParcel(parcel: Parcel): FnSentencePointer {
                return FnSentencePointer(parcel)
            }

            override fun newArray(size: Int): Array<FnSentencePointer?> {
                return arrayOfNulls(size)
            }
        }
    }
}
